doc/user/content/sql/functions/filters.md

```diff
@@ -17,6 +19,7 @@ You can use a `FILTER` clause on an aggregate function to specify which rows are
 ```sql
 SELECT
     COUNT(*) AS unfiltered,
-    COUNT(*) FILTER (WHERE i < 5) AS filtered
+    -- The FILTER guards the evaluation which might otherwise error.
+    COUNT(1 / (5 - i)) FILTER (WHERE i < 5) AS filtered
```

Comment:
Nikhil says here that we don't need to make it guard errors: https://materializeinc.slack.com/archives/C02CB7L4TCG/p1691987351125739?thread_ts=1691752269.450799&cid=C02CB7L4TCG

Also, in general, we might want to hold off on advertising that CASE WHEN (which FILTER translates to) can guard an error, because we have several hard-to-fix issues around this, see [here](https://github.com/MaterializeInc/materialize/issues/17189) at "CASE WHEN".